docs/advanced-usage.rst

```diff
@@ -0,0 +1,176 @@
+Advanced Usage
+==============
+
+.. currentmodule:: urllib3
+
+.. _stream:
+
+Streaming and IO
+----------------
+
+When dealing with large responses it's often better to stream the response
+content::
+
+    >>> r = http.request(
+    ...     'GET',
+    ...     'http://httpbin.org/bytes/1024',
+    ...     preload_content=False)
+    >>> for chunk in r.stream(32):
+    ...     print(chunk)
+    b'...'
+    b'...'
+    ...
+    >>> r.release_conn()
+
+Setting ``preload_content`` to ``False`` means that urllib3 will stream the
+response content. :meth:`~response.HTTPResponse.stream` lets you iterate over
+chunks of the response content.
+
+.. note:: When using ``preload_content=False``, you should call 
+    :meth:`~response.HTTPResponse.release_conn` to release the http connection
+    back to the connection pool so that it can be re-used.
+
+However, you can also treat the :class:`~response.HTTPResponse` instance as
+a file-like object. This allows you to do buffering::
+
+    >>> import io
```

Comment:
I mean, I presume its imported for the *next* code block, but I'd rather see that delayed.